src/rust/ide/view/src/documentation.rs

```diff
@@ -148,7 +160,37 @@ impl ViewModel {
     /// Create a container for generated content and embed it with stylesheet.
     fn push_to_dom(&self, content:String) {
         let data_str = format!(r#"<div class="docVis">{}{}</div>"#,documentation_style(),content);
-        self.dom.dom().set_inner_html(&data_str)
+        self.dom.dom().set_inner_html(&data_str);
+    }
+
+    /// Append listeners to copy buttons in doc to enable copying examples.
+    fn append_listeners_to_copy_buttons(&self) {
+        let code_blocks  = self.dom.dom().get_elements_by_class_name("CodeBlock");
+        let copy_buttons = self.dom.dom().get_elements_by_class_name("copyCodeBtn");
+        for i in 0..copy_buttons.length() {
+            let copy_button = copy_buttons.get_with_index(i);
+            let code_block  = code_blocks.get_with_index(i);
+            match copy_button {
+                Some(btn) => match code_block {
+                    Some(code) => {
+                        let cpy_btn  = btn.dyn_into::<HtmlElement>();
+                        let code_blk = code.dyn_into::<HtmlElement>();
+                        let closure  = move |_event: MouseEvent| {
+                            copyCode(&code_blk.unwrap());
+                        };
+                        let closure  = Closure::wrap(Box::new(closure).into_fn_mut());
+                        let callback = closure.as_ref().unchecked_ref();
+                        match cpy_btn.unwrap().add_event_listener_with_callback("click", callback){
+                            Ok(_)    => (),
+                            Err(err) => error!(&self.logger, "Unable to add event listener: {err:?}")
+                        }
+                        closure.forget();
+                    },
+                    None => info!(&self.logger, "No code block."),
+                },
+                None => info!(&self.logger, "No copy button."),
```

Comment:
This info is not descriptive at all. If I see a message in a console "No code block" I would not understand it at all. Moreover, should it be "info"? Is it something you want the user to be informed about? Isnt this a debug information or a warning?